zerver/tests/test_custom_profile_data.py

```diff
@@ -190,17 +231,32 @@ def test_update_profile_data(self) -> None:
             if f['id'] == field.id:
                 self.assertEqual(f['value'], 'foobar')
 
+    def test_update_choice_field(self) -> None:
+        self.login(self.example_email("iago"))
+        realm = get_realm('zulip')
+        field = CustomProfileField.objects.get(name='Favorite character',
+                                               realm=realm)
+        data = [{
+            'id': field.id,
+            'value': 'foobar',
+        }]
+
+        result = self.client_patch("/json/users/me/profile_data",
+                                   {'data': ujson.dumps(data)})
+        self.assert_json_error(result,
+                               "'foobar' is not a valid choice for 'value[4]'.")
+
```

Comment:
This should have a passing test after this, right?

Reply:
Done.